restli-server/src/main/java/com/linkedin/restli/internal/server/ResourceContextImpl.java

```diff
@@ -184,31 +196,42 @@ public ResourceContextImpl(final MutablePathKeys pathKeys,
 
     TimingContextUtil.beginTiming(requestContext, FrameworkTimingKeys.SERVER_REQUEST_RESTLI_PROJECTION_DECODE.key());
 
-    if (_parameters.containsKey(RestConstants.FIELDS_PARAM))
+    LocalProjectionMaskAttr localProjectionMaskAttr =
+        (LocalProjectionMaskAttr) requestContext.getLocalAttr(CONTEXT_PROJECTION_MASKS_KEY);
+    if (localProjectionMaskAttr != null)
     {
-      _projectionMask = ArgumentUtils.parseProjectionParameter(getParameter(RestConstants.FIELDS_PARAM));
+      _projectionMask = localProjectionMaskAttr.getProjectionMask();
+      _metadataProjectionMask = localProjectionMaskAttr.getMetadataProjectionMask();
+      _pagingProjectionMask = localProjectionMaskAttr.getPagingProjectionMask();
     }
     else
     {
-      _projectionMask = null;
-    }
+      if (_parameters.containsKey(RestConstants.FIELDS_PARAM))
+      {
+        _projectionMask = ArgumentUtils.parseProjectionParameter(getParameter(RestConstants.FIELDS_PARAM));
+      }
+      else
+      {
+        _projectionMask = null;
+      }
 
-    if (_parameters.containsKey(RestConstants.METADATA_FIELDS_PARAM))
-    {
-      _metadataProjectionMask = ArgumentUtils.parseProjectionParameter(getParameter(RestConstants.METADATA_FIELDS_PARAM));
-    }
-    else
-    {
-      _metadataProjectionMask = null;
-    }
+      if (_parameters.containsKey(RestConstants.METADATA_FIELDS_PARAM))
```

Comment:
Is there inconsistent tab/space uses this line? (github diff seems to have highlighted this)

Reply:
It looks ok in the IDE and in local git diff... Not sure why.